packages/pipeline-services/src/migration/migrateV1/index.ts

```diff
@@ -14,46 +14,46 @@
  * limitations under the License.
  */
 
-function migrate(pipeline: any) {
-  if (pipeline.pipelines[0].app_data) {
-    // title -> name
-    pipeline.pipelines[0].app_data.name = pipeline.pipelines[0].app_data.title;
-    delete pipeline.pipelines[0].app_data.title;
-
-    delete pipeline.pipelines[0].app_data.export;
-    delete pipeline.pipelines[0].app_data.export_format;
-    delete pipeline.pipelines[0].app_data.export_path;
+const hasAppdataField = (node: any, fieldName: string): boolean => {
+  return !!node.app_data && !!node.app_data[fieldName];
+};
 
-    pipeline.pipelines[0].app_data.version = 1;
-  } else {
-    pipeline.pipelines[0].app_data = { version: 1 };
+const deleteAppdataField = (node: any, fieldName: string): void => {
+  if (hasAppdataField(node, fieldName)) {
+    delete node.app_data[fieldName];
   }
+};
+
+const renameAppdataField = (
+  node: any,
+  currentFieldName: string,
+  newFieldName: string
+): void => {
+  if (hasAppdataField(node, currentFieldName)) {
+    node.app_data[newFieldName] = node.app_data[currentFieldName];
+    deleteAppdataField(node, currentFieldName);
+  }
+};
```

Comment:
I'd rather leave the refactoring out of this PR. It's hard for me to tell what bug you are fixing if any. 

I would prefer to avoid extra functions with side effects, but if we do want to refactor this because it becomes too much boilerplate per rename with your fixes (is the fix checking that the old property is defined?), then I would prefer something closer to `renameField(obj, { from: "old-name", to: "new-name" })`